memory/Array.h

```diff
@@ -0,0 +1,86 @@
+#pragma once
+
+namespace memory
+{
+
+template <typename T, size_t SIZE>
+class Array
+{
+public:
+    typedef const T* const_iterator;
+    typedef T* iterator;
+
+    Array() : _capacity(SIZE), _size(0), _dataPtr(reinterpret_cast<T*>(_data)) {}
+    explicit Array(size_t size) : _capacity(size), _size(0), _dataPtr(reinterpret_cast<T*>(_data))
+    {
+        if (size > SIZE)
+        {
+            _dataPtr = reinterpret_cast<T*>(malloc(size * sizeof(T)));
```

Comment:
[minor] malloc will allocate memory aligned to `std::max_align_t` (which I guess is 16 byte) which should work well for all our cases. But we have a better function to allocate this with alignment specification so we don't need a strict align to 16 when we are storing bytes.

`std::aligned_alloc(std::alignment_of<T>::value, size * sizeof(T))`